internal/istio/client.go

```diff
@@ -68,34 +63,6 @@ func (c *Client) ListVirtualServices(ctx context.Context, namespace string) (*is
 	return virtualServiceList, nil
 }
 
```

Comment:
My intention was to separate the code concerned with the object itself (i.e. the construction in this case) from the client that is using the objects. Personally, I am preferring such separation as it makes it easier to understand what code deals with what concern. If you see it differently I am happy to discuss.

Thinking about it, I also noticed that `Client` in `istio/client.go` is as of now also not really an "IstioClient". Maybe it is worth also renaming this to `VirtualServiceClient`. But this is a slightly different discussion.

Reply:
Yeah mean like in the following?

```
- istio
  - resource
    - ...
  - istioclient.go
    - # Client struct
    - # NewVersionedIstioClient
  - virtualserviceclient.go
    - # GetVirtualService
    - # ListVirtualService
    - ...
```

It sounds doable (because of the use of receiver functions), but I unfortunately lack the experience whether this is common practice or not in go. I will try to check :)

Reply:
I thought about it some time and moved the virtual service part out of `resources` again and made it part of the `istio` package directly. The API surface of `istio` package is still fairly small. If in the future we see that it is growing too big, we can introduce sub-packages.

Wrt. to moving the CRUD functions to a virtual service client: I looked a bit closer at the client and was able to carve out further functions that did not really belong to client. Now we have a couple of VirtualService related methods and one method related to gateways. For simplicity, I opted now for keeping it a `istio` level client since it includes VS and GW related methods. We could of course split it into two clients but it may be a bit overkill to handle two client dependencies. Also the resulting client file is now fairly small and concise, so it would be okay for me to keep it as is. But if you prefer to move the receiver functions out to separate files I can also do so